### PR TITLE
Because SymEngine 0.10, undoing 8456

### PR DESCRIPTION
**Pull request by 1ucian0**
[Waiting for SymEngine 0.10](https://pypi.org/project/symengine/) ([milestone](https://github.com/symengine/symengine.py/milestone/2)) to be released, this PR undoes https://github.com/Qiskit/qiskit-terra/pull/8456/files

### CODE REVIEW

**Comment by nkanazawa1989** [comment]
Similar issue and https://github.com/Qiskit/qiskit-terra/pull/8232 is also waiting for symengine 0.10.

**Review by nkanazawa1989 — APPROVED** [review]
Nice! Finally we get symengine 0.10 :)

File: qiskit/circuit/parameterexpression.py [hunk 1]
```diff
@@ -521,24 +521,17 @@ def __eq__(self, other):
...
-        if not symbol_expr.is_real and symbol_expr.is_real is not None:
+        if not self._symbol_expr.is_real and self._symbol_expr.is_real is not None:
             # Symengine returns false for is_real on the expression if
             # there is a imaginary component (even if that component is 0),
             # but the parameter will evaluate as real. Check that if the
             # expression's is_real attribute returns false that we have a
             # non-zero imaginary
             if _optionals.HAS_SYMENGINE:
-                if symbol_expr.imag == 0.0:
+                if self.symbol_expr.imag == 0.0:
```
Comment (Cryoris) [inline qiskit/circuit/parameterexpression.py:531]:
```suggestion
                if self._symbol_expr.imag == 0.0:
```

File: qiskit/circuit/parameterexpression.py [hunk 2]
```diff
@@ -521,24 +521,17 @@ def __eq__(self, other):
...
                     return True
             return False
-        return symbol_expr.is_real
+        return self.symbol_expr.is_real
```
Comment (Cryoris) [inline qiskit/circuit/parameterexpression.py:534]:
```suggestion
        return self._symbol_expr.is_real
```

**Comment by nkanazawa1989** [comment]
An thoughts? @1ucian0 
https://github.com/Qiskit/qiskit-terra/pull/9857#pullrequestreview-1358806062

**Comment by 1ucian0** [comment]
`on hold` until we decide what to do with py37 support (see https://github.com/Qiskit/qiskit-terra/pull/9857#pullrequestreview-1358806062)

**Comment by 1ucian0** [comment]
symengine [updated their wheels to have py37](https://github.com/symengine/symengine-wheels/commit/f6c52a720b4aa7b97e8bfc00ea72fd9d67ffb0a4#commitcomment-106083238). So removing `on hold`

**Pull request closed without merging**